test/test-directives.js

```diff
@@ -341,62 +341,56 @@ Test('allOf', function (t) {
         t.ok(!schema.validate({ a: 'string', b: 10 }).error, 'no error');
         const validation = schema.validate({ a: 'string', b: 'string' });
         t.equal(validation.error.name, 'ValidationError', 'error');
-        t.equal(validation.error.message, '"b" must be a number', 'error');
+        t.equal(validation.error.details[0].type, 'alternatives.all', 'error');
     });
 
-    t.test('allOf array with different types', function (t) {
+    t.test('oneOf array with different types', function (t) {
         t.plan(1);
 
-        // This validates because array elements can be of different types and similar
-        // to an object where schemas are additive schemas are additive here too
         const schema = Enjoi.schema({
-            'allOf': [
-                {
-                    type: 'array',
-                    items: [
-                        {
-                            type: 'string'
-                        }
-                    ]
-                },
-                {
-                    type: 'array',
-                    items: [
-                        {
-                            type: 'number'
-                        }
-                    ]
-                }
-            ]
+            type: 'array',
+            items: {
+                oneOf: [
+                    { type: 'string' },
+                    { type: 'number' }
+                ]
+            }
         });
 
-        t.ok(!schema.validate(['string', 10]).error, 'error');
+        t.ok(!schema.validate(['string', 10]).error, 'no error');
     });
 
-    t.test('allOf object with conflicting requirements', function (t) {
```

Comment:
removed this test since allOf using `alternatives.match('all')` can compile this